app/src/cc/arduino/contributions/ui/FilterJTextField.java

```diff
@@ -101,13 +101,14 @@ private void spawnTimer() {
   }
 
   public void applyFilter() {
-    String filter = showingHint ? "" : getText();
-    filter = filter.toLowerCase();
-
-    // Replace anything but 0-9, a-z, or : with a space
-    filter = filter.replaceAll("[^\\x30-\\x39^\\x61-\\x7a^\\x3a]", " ");
-
-    onFilter(filter.split(" "));
+    if (!showingHint) {
+      String filter = getText().toLowerCase();
+        
+      // Replace anything but 0-9, a-z, or : with a space
+      filter = filter.replaceAll("[^\\x30-\\x39^\\x61-\\x7a^\\x3a]", " ");
+      
+      onFilter(filter.split(" "));
```

Comment:
As I said in the other comment this method should be called with an empty array if `showingHint` is `false`.

Reply:
Thanks for the note, I have addressed the above and pushed the changes.